stdlib/public/Concurrency/Task.cpp

```diff
@@ -1394,6 +1394,65 @@ swift_task_createNullaryContinuationJobImpl(
   return job;
 }
 
+/// Job that allows to use executor API to schedule a block of task-less
+/// synchronous code.
+class AdHocJob : public Job {
+
+private:
+  void *Context;
+  AdHocWorkFunction *Work;
+  TaskLocal::Storage Local;
+
+public:
+  AdHocJob(JobPriority priority, void *context, AdHocWorkFunction *work)
+      : Job({JobKind::AdHoc, priority}, &process), Context(context),
+        Work(work) {
+    TaskLocal::copyTo(&Local, nullptr);
```

Comment:
For GAIT, I expect that vast majority of releases to happen within isolating actor, so I'm not very concerned about performance costs of the slow path. But for actors, that's probably not true. Maybe what you mentioned [below](https://github.com/apple/swift/pull/60057#discussion_r922585619) can help to reduce switching for actors.

Reply:
I would not expect to have task locals available in a deinit, and especially since we don't guarantee WHICH exact locals those will be, so basing any tracing infra on "object got deallocated and must have right locals" is going to be wrong at some point so not usable.

I'd say losing the locals is expected. For anything that needs things at "end of scope" (like tracing) the usual pattern is to wrap in a closure, not really rely on `deinit`.

Reply:
I think some task-locals can still be useful in the `deinit` even if you are not sure which task/thread performs the last release. For example, if you have a logger applied to entire app or a large subsystem, you can expect that it will be available in any task/thread launched within that scope. So it does not matter which one makes the last release, as long you are confident that last release happens within this coarse-grained scope, you can still meaningfully use logger's task local in the `deinit`.